frontend/ui/elements/screen_keep_on_menu_table.lua

```diff
@@ -0,0 +1,16 @@
+local isAndroid, android = pcall(require, "android")
+local _ = require("gettext")
+
+if not isAndroid then return end
+
+return {
+    text = _("Keep screen on"),
+    checked_func = function()
+        return G_reader_settings:isTrue("keep_screen_on")
+    end,
+    callback = function()
+        local keep_screen_on = G_reader_settings:isTrue("keep_screen_on")
+        android.setKeepScreenOn(not keep_screen_on)
```

Comment:
Have you tried just doing it a couple of times here? E.g., the same line twice or 

```
android.setKeepScreenOn(not keep_screen_on)
android.setKeepScreenOn(keep_screen_on)
android.setKeepScreenOn(not keep_screen_on)
```

Sure, it'd be ugly, but if it works around the problem worse things have happened. ;-)

Reply:
It seems a bad idea, it breaks the app on both devices. :astonished: